### PR TITLE
Stress test a worker uploading large bundle results

### LINKED ISSUE #3053: Stress test a worker uploading large bundle results
**Describe the bug**
Add a stress test for workers uploading large bundle results.

### PR DESCRIPTION
**Pull request by teetone**
### Reasons for making this change

Add a stress test that uploads a large amount of data after the bundle finishes running.

### Related issues

Resolves #3053

### CODE REVIEW

File: tests/stress/stress_test.py [hunk 1]
```diff
@@ -326,6 +366,9 @@ def main():
 
     if args.heavy:
         print('Setting the heavy configuration...')
+        # Set the sizes of the large files to be bigger than the max memory on the system to test that data
+        # is being streamed when the large bundles are used as a dependencies.
+        args.large_dependency_size_gb = 16
```
Comment (yiboliu) [inline tests/stress/stress_test.py:371]: Do we have somewhere specifying the max size of memory? If yes, I think we could probably not hard-code here.

File: tests/stress/stress_test.py [hunk 2]
```diff
@@ -169,10 +185,34 @@ def _heartbeat(self):
             # Have heartbeat run every 30 seconds
             time.sleep(30)
 
-    def _test_large_bundle(self):
-        self._set_worksheet('large_bundles')
-        # Set this to larger than the max memory on the system to test that data is being
-        # streamed when the large bundle is being used as a dependency.
+    def _test_large_bundle_result(self):
+        def create_large_file_in_bundle(large_file_size_gb):
```
Comment (yiboliu) [inline tests/stress/stress_test.py]: type annotation plz

File: tests/stress/stress_test.py [hunk 3]
```diff
@@ -169,10 +185,34 @@ def _heartbeat(self):
...
+            )
+            return TestFile('large_dependency.py', content=code)
+
+        self._set_worksheet('large_bundle_result')
+        file = create_large_file_in_bundle(self._args.large_dependency_size_gb)
+        self._run_bundle([self._cl, 'upload', file.name()])
+        file.delete()
+
+        dependency_uuid = self._run_bundle(
+            [self._cl, 'run', ':' + file.name(), 'python ' + file.name()]
```
Comment (yiboliu) [inline tests/stress/stress_test.py:201]: Is the file created a python file?

File: tests/stress/stress_test.py (hunk 1 above)
Comment (teetone) [reply tests/stress/stress_test.py:371]: No, there isn't an easy way to get that number.

File: tests/stress/stress_test.py (hunk 3 above)
Comment (teetone) [reply tests/stress/stress_test.py:201]: Yes

File: tests/stress/stress_test.py (hunk 2 above)
Comment (teetone) [reply tests/stress/stress_test.py]: Thanks for the reminder :)

**Review by epicfaace — APPROVED** [review]

**Pull request merged**